src/server/job_func.c

```diff
@@ -767,6 +778,24 @@ job_purge(job *pjob)
 	if (pjob->ji_mjspipe2 != -1) {
 		(void)close(pjob->ji_mjspipe2);
 	}
+
+	/* if open, close 3rd pipes to/from Mom starter process */
+	if (pjob->ji_child2parent_job_update_pipe != -1) {
+		(void)close_conn(pjob->ji_child2parent_job_update_pipe);
+	}
+	if (pjob->ji_parent2child_job_update_pipe != -1) {
+		(void)close(pjob->ji_parent2child_job_update_pipe);
+	}
```

Comment:
Don't need to check the value before closing since you're casting to void. Probably want to set the value to -1 after you close.